### PR TITLE
fix(2074): Use admin permission to sync hook

### LINKED ISSUE #2074: Cannot sync webhook
**What happened**:
Click the sync webhook button in the pipeline option tub, it shows `Not found` error message and its webhook is not synced.

**What you expected to happen**:
Can sync webhook or shows appropriate error message.

**How to reproduce it**:
1. A user create an pipeline.
2. Change permission of the user to write permission in the GitHub.
3. Click sync webhook button in the pipeline option tub. 

This is because of the difference between the pipeline admin permission and hook edit permission.
To start pipeline, write permission is sufficient.
Therefore, Screwdriver checks the admin to see if the user has push permissions.
https://github.com/screwdriver-cd/models/blob/56e3ce9b02c68b99ca5848f3e9e029702f742f11/lib/pipeline.js#L954
However, to edit repository hook, admin permission is needed in the GitHub.
https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization

### PR DESCRIPTION
**Pull request by yoshwata**
## Context

fix https://github.com/screwdriver-cd/screwdriver/issues/2074

## Objective

This PR uses admin permission to sync hook, so the `Not Found` error doesn't occur anymore.

For the case there is no admin permission, it shows error message like below.
Screen Shot 2020-06-11 at 16 42 19

## References

https://github.com/screwdriver-cd/screwdriver/issues/2074

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: lib/pipeline.js [hunk 1]
```diff
@@ -378,8 +378,11 @@ class PipelineModel extends BaseModel {
      * @method  addWebhook
      * @return  {Promise}
      */
-    addWebhook(webhookUrl) {
-        return this.token.then(token =>
+    async addWebhook(webhookUrl) {
+        const admin = await this.getFirstRepoAdmin();
+
+        // return this.token.then(token =>
```
Comment (kumada626) [inline lib/pipeline.js]: Remove comment.

File: lib/pipeline.js [hunk 2]
```diff
@@ -378,8 +378,11 @@ class PipelineModel extends BaseModel {
      * @method  addWebhook
      * @return  {Promise}
      */
-    addWebhook(webhookUrl) {
-        return this.token.then(token =>
+    async addWebhook(webhookUrl) {
+        const admin = await this.getFirstRepoAdmin();
```
Comment (wahapo) [inline lib/pipeline.js]: I think it is better to implement it without using await.

**Review by wahapo — APPROVED** [review]
lgtm

File: lib/pipeline.js [hunk 3]
```diff
@@ -379,14 +379,16 @@ class PipelineModel extends BaseModel {
...
     addWebhook(webhookUrl) {
-        return this.token.then(token =>
-            this.scm.addWebhook({
-                scmUri: this.scmUri,
-                scmContext: this.scmContext,
-                token,
-                webhookUrl
-            })
-        );
+        return this.getFirstRepoAdmin()
```
Comment (jithine) [inline lib/pipeline.js:382]: do we have checks before reaching here that the user has atleast write permissions ? If not we might end up allowing anyone to add webhook.

Comment (yoshwata) [reply lib/pipeline.js:382]: Thanks. Write permission is checked at [this code](https://github.com/screwdriver-cd/screwdriver/blob/ce6822c9c9603f1c6075365ff02eea9a33824d16/plugins/pipelines/syncWebhooks.js#L47).

File: test/lib/pipeline.test.js [hunk 4]
```diff
@@ -380,6 +380,19 @@ describe('Pipeline Model', () => {
             });
         });
 
+        it('rejects if there is no admins', () => {
+            // scmMock.addWebhook.rejects(new Error('error adding webhooks'));
```
Comment (kumada626) [inline test/lib/pipeline.test.js]: Should remove this line.

**Comment by kumada626** [comment]
Small comment. Otherwise, LGTM.

**Review by kumada626 — APPROVED** [review]

**Review by jithine — APPROVED** [review]
This PR branch needs to be rebased before merging.

**Review by tkyi — APPROVED** [review]

**Pull request merged**